pkg/engine/constraints/node_constraint.go

```diff
@@ -0,0 +1,50 @@
+package constraints
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+)
+
+type (
+	// NodeConstraint is a struct that represents constraints that can be applied on a specific node in the resource graph.
+	// NodeConstraints are used to control intrinsic properties of a node in the resource graph
+	NodeConstraint struct {
+		Operator ConstraintOperator `yaml:"operator"`
+		Target   core.ResourceId    `yaml:"target"`
+		Property string             `yaml:"property"`
+		Value    any                `yaml:"value"`
+	}
+)
+
+func (b *NodeConstraint) Scope() ConstraintScope {
+	return EdgeConstraintScope
+}
+
+func (b *NodeConstraint) IsSatisfied(dag *core.ResourceGraph) bool {
+	switch b.Operator {
+	case EqualsConstraintOperator:
+		res := dag.GetResource(b.Target)
+		if res == nil {
+			return false
+		}
+		val := reflect.ValueOf(res).Elem().FieldByName(b.Property)
+		return val.Interface() == b.Value
+	}
+	return false
+}
+
+func (b *NodeConstraint) Conflict(other Constraint) bool {
+	return false
+}
+
+func (b *NodeConstraint) Validate() error {
+	if b.Target.Provider == core.AbstractConstructProvider {
+		return errors.New("node constraint cannot be applied to an abstract construct")
+	}
+	if b.Property == "" || reflect.ValueOf(b.Value).IsZero() {
```

Comment:
Does this imply I can't say something like "set number of replicas to 0"?

Reply:
right now it does imply that, which maybe is incorrect, so maybe i should remove the check of value being a zero value

Reply:
👍 Yeah, I would. If we later want to ensure some value isn't 0, a system-provided constraint can do that ("instanceCount must be > 0", which would then be in conflict with "instanceCount = 0" if the user provided that).